builtin/aws/alb/releaser.go

```diff
@@ -42,192 +46,357 @@ func (r *Releaser) StatusFunc() interface{} {
-		lbName = "waypoint-" + src.App
+	lbName string,
+	port int64,
+	state *Resource_SecurityGroup,
+) error {
+	if r.config.ListenerARN != "" {
+		// a custom listner is being used, so we assume the Load Balancer is already
+		// created, and alrady has a security group associated with it.
+		log.Debug("custom listener arn found, skipping security group create")
+		return nil
 	}
 
-	// We have to clamp at a length of 32 because the Name field to
-	// CreateLoadBalancer requires that the name is 32 characters or less.
-	if len(lbName) > 32 {
-		lbName = lbName[:32]
-		log.Debug("using a shortened value for load balancer name due to AWS's length limits", "lbName", lbName)
-	}
+	ec2srv := ec2.New(sess)
 
+	// Figure out what subnets we're using. This will give us the VPC
+	// information.
 	var (
-		certs    []*elbv2.Certificate
-		protocol string = "HTTP"
-		port     int64  = 80
+		vpc     *string
+		subnets []*string
 	)
+	if len(r.config.Subnets) == 0 {
+		_, v, err := utils.DefaultSubnets(ctx, sess)
+		if err != nil {
+			return err
+		}
 
-	if r.config.CertificateId != "" {
-		protocol = "HTTPS"
-		port = 443
-		certs = append(certs, &elbv2.Certificate{
-			CertificateArn: &r.config.CertificateId,
+		vpc = v
+	} else {
+		for _, s := range r.config.Subnets {
+			subnets = append(subnets, aws.String(s))
+		}
+
+		subnetInfo, err := ec2srv.DescribeSubnets(&ec2.DescribeSubnetsInput{
+			SubnetIds: []*string{subnets[0]},
 		})
-	}
+		if err != nil {
+			return err
+		}
 
-	// If there is a port defined, honor it.
-	if r.config.Port != 0 {
-		port = int64(r.config.Port)
+		vpc = subnetInfo.Subnets[0].VpcId
 	}
+	sGroup, err := utils.CreateSecurityGroup(ctx, sess, fmt.Sprintf("%s-incoming", lbName), vpc, int(port))
+	if err != nil {
+		return err
+	}
+	state.Id = *sGroup
 
-	var (
-		lb               *elbv2.LoadBalancer
-		listener         *elbv2.Listener
-		newListener      bool
-		existingListener string
-	)
+	return nil
+}
 
-	if r.config.ListenerARN != "" {
-		existingListener = r.config.ListenerARN
+func (r *Releaser) resourceSecurityGroupDestroy(
+	ctx context.Context,
+	sess *session.Session,
+	state *Resource_SecurityGroup,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+) error {
+	if state.Id != "" {
+		step := sg.Add("Destroying Security Group...")
+		defer step.Abort()
+		log.Debug("deleting security group", "sg-id", state.Id)
+		ec2Svc := ec2.New(sess)
+		for i := 0; i < 20; i++ {
+			_, err := ec2Svc.DeleteSecurityGroup(&ec2.DeleteSecurityGroupInput{
+				GroupId: &state.Id,
+			})
+			if err == nil {
+				step.Done()
+				return nil
+			}
+			// if we encounter an unrecoverable error, exit now.
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case "DependencyViolation":
+					time.Sleep(2 * time.Second)
+					continue
+				case "InvalidGroup.NotFound":
+					log.Debug("security group not found", "sg-id", state.Id)
+					return nil
+				default:
+					return err
+				}
+			}
+			return err
+		}
+		step.Done()
```

Comment:
Could use a "finished" update after the destroy

```suggestion
		step.Update("Destroyed Security Group!")
		step.Done()
```